test/e2e/helpers.go

```diff
@@ -93,11 +90,40 @@ func CreateK8sObjectWithRetry[T kruntime.Object](
 // and the parameters for it. It then makes the call with some retry logic.
 func DeleteK8sObjectWithRetry(
 	ctx context.Context, deleteFunc K8sDeleteFunc, name string, options metav1.DeleteOptions,
-) {
+) (err error) {
```

Comment:
I expanded this to return an error so I could match on a NotFound in the Cleanup. An alternative I considered was to have a second Delete function that would return the error but returning the error seemed like the more "Go" thing to do even though it introduced all these underscores.

What do people think?